src/filter.cpp

```diff
@@ -140,73 +140,92 @@ namespace vg{
-
-
-        if (aln_first.name() == "" || aln_first.name() == ""){
-            return make_pair(Alignment(), Alignment());
-        }
-        else{
-            return make_pair(aln_first, aln_second);
-        }
-
-    }
-    pair<Alignment, Alignment> Filter::soft_clip_filter(Alignment& aln_first, Alignment& aln_second){
-        Alignment a_check = soft_clip_filter(aln_first);
-        Alignment b_check = soft_clip_filter(aln_second);
-
-        if (a_check.name() == "" || b_check.name() == ""){
-            return inverse ? make_pair(Alignment(), Alignment()) : make_pair(aln_first, aln_second) ;
-        }
-        else{
-            return inverse ? make_pair(aln_first, aln_second) : make_pair(Alignment(), Alignment()) ;
-        }
-
-    }
-    pair<Alignment, Alignment> split_read_filter(Alignment& aln_first, Alignment& aln_second){
-
-    }
-    pair<Alignment, Alignment> path_divergence_filter(Alignment& aln_first, Alignment& aln_second){
-
-    }
-    pair<Alignment, Alignment> reversing_filter(Alignment& aln, Alignment& aln_second){
-
-    }
+    // std::pair<Alignment, Alignment> Filter::path_length_filter(Alignment& aln_first, Alignment& aln_second){
+    //     Alignment x = path_length_filter(aln_first);
+    //     Alignment y = path_length_filter(aln_second);
+    //     if (x.name().empty() || y.name().empty()){
+    //         return inverse ? make_pair(x, y) : make_pair(Alignment(), Alignment());
+    //     }
+    //     else{
+    //         return  inverse ? make_pair(Alignment(), Alignment()) : make_pair(x, y);
+    //     }
+    // }
+
+
+
+
+    // pair<Alignment, Alignment> Filter::depth_filter(Alignment& aln_first, Alignment& aln_second){
+    //     aln_first = depth_filter(aln_first);
+    //     aln_second = depth_filter(aln_second);
+    //     if (!(aln_first.name() == "") && !(aln_first.name() == "")){
+    //         return inverse ? make_pair(aln_first, aln_second) : make_pair(Alignment(), Alignment());
+    //     }
+    //     else{
+    //         return inverse ? make_pair(Alignment(), Alignment()) : make_pair(aln_first, aln_second);
+    //     }
+    // }
+    // pair<Alignment, Alignment> qual_filter(Alignment& aln_first, Alignment& aln_second){
+
+
+    //     if (aln_first.name() == "" || aln_first.name() == ""){
+    //         return make_pair(Alignment(), Alignment());
+    //     }
+    //     else{
+    //         return make_pair(aln_first, aln_second);
+    //     }
+
+    // }
+    // pair<Alignment, Alignment> percent_identity_filter(Alignment& aln_first, Alignment& aln_second){
+
+
+    //     if (aln_first.name() == "" || aln_first.name() == ""){
+    //         return make_pair(Alignment(), Alignment());
+    //     }
+    //     else{
+    //         return make_pair(aln_first, aln_second);
+    //     }
+
+    // }
+    // pair<Alignment, Alignment> Filter::soft_clip_filter(Alignment& aln_first, Alignment& aln_second){
```

Comment:
I think I'm going to merge it into whole-lotta-genotype and just bring everything in at once, rather than multiple PRs having to get brought in.